src/docs/cookbook/navigation/named-routes-drawer.md

```diff
@@ -0,0 +1,236 @@
+---
+title: Navigate with named routes within a drawer
+---
+
+In the
+[Navigate with named routes](/docs/cookbook/navigation/named-routes/)
+recipe, we learned how to Navigate to the same screen in many parts of our apps
+via a "named route". In the
+[Add a Drawer to a screen](/docs/cookbook/design/drawer/) recipe, we
+learned how to user a drawer for Navigation.
+
+This example will replicate the functionality from the original two recipes,
+demonstrating how to combine named routes with a drawer to Navigate between
+pages at the root of an application.
+
+## Directions
+
+  1. Create a Drawer
+  1. Create two screens
+  3. Define the routes
+
+## 1. Create a Drawer
+
+In order to share Navigation logic used in the drawer between screens we'll
+create a `NavigationDrawer` widget. Within the `onTap` of our `ListTitle`
+widgets we'll use the
+[`Navigator.pushReplacementNamed`](https://docs.flutter.io/flutter/widgets/Navigator/pushReplacementNamed.html)
+function. This associates the list items in our `Drawer` with the screens we'll
+define our `routes` table.
+
+<!-- skip -->
+```dart
+class NavigationDrawer extends StatelessWidget  {
+  @override
+  Widget build(BuildContext context) {
+    // Create the Drawer which defines Navigation logic we'll reuse in each
+    // widget residing at the root of our application.
+    return Drawer(
+      // Add a ListView to the drawer. This ensures the user can scroll
+      // through the options in the Drawer if there isn't enough vertical
+      // space to fit everything.
+      child: ListView(
+        // Important: Remove any padding from the ListView.
+        padding: EdgeInsets.zero,
+        children: <Widget>[
+          DrawerHeader(
+            child: Text('Drawer Header'),
+            decoration: BoxDecoration(
+              color: Colors.blue,
+            ),
+          ),
+          ListTile(
+            title: Text('First'),
+            onTap: () {
+              // Update the state of the app
```

Comment:
Rather than only keeping these comments, perhaps we could say something about why you're using `pushReplacement` here and what effect that will have?